src/hal/src/buffer.rs

```diff
@@ -88,6 +88,21 @@ bitflags!(
     }
 );
 
+bitflags!(
+    /// Buffer create flags.
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub struct CreateFlags: u32 {
```

Comment:
I think we should move them out into `memory::SparseFlags` and use as an extra argument in both `create_buffer` and `create_image` for the following reasons:
- the flags are the same for both buffers and images
- the image `ViewCapabilities` is meant for outlining restrictions on how the views are created. Trying to fit the sparse flags in there does not match the old meaning of the flags.